### PR TITLE
ST board Id: Alignment with pyOCD v0.30.3

### PR DESCRIPTION
**Pull request by jeromecoutant**
### Description

Alignment with pyOCD v0.30.3

Same action done on pyOCD: https://github.com/pyocd/pyOCD/pull/1166

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by jeromecoutant** [comment]
@MarceloSalazar

File: src/mbed_os_tools/detect/platform_database.py [hunk 1]
```diff
@@ -165,16 +166,20 @@
...
         u"0840": u"B96B_F446VE",
         u"0841": u"NUCLEO_G474RE",
+        u"0842": u"NUCLEO_H753ZI",
         u"0843": u"NUCLEO_H745ZI_Q",
         u"0847": u"DISCO_H745I",
         u"0849": u"NUCLEO_G070RB",
         u"0850": u"NUCLEO_G431RB",
+        u"0851": u"NUCLEO_G431KB",
         u"0852": u"NUCLEO_G031K8",
+        u"0853": u"NUCLEO_F031K8",
```
Comment (MarceloSalazar) [inline src/mbed_os_tools/detect/platform_database.py:176]: This seems a typo. It should be `NUCLEO_F301K8`

**Review by MarceloSalazar — CHANGES_REQUESTED** [review]
Please fix the target name

**Comment by MarceloSalazar** [comment]
@Patater can you please merge when you have chance?

**Comment by jeromecoutant** [comment]
ping

**Comment by MarceloSalazar** [comment]
@Patater please review and merge - thanks

**Pull request merged**